docx/section.py

```diff
@@ -37,11 +38,25 @@ class Section(object):
     """
     Document section, providing access to section and page setup settings.
     """
+    _header = None
+
     def __init__(self, sectPr):
         super(Section, self).__init__()
         self._sectPr = sectPr
 
     @property
+    def header(self):
+        if not self._header:
+            # here's how I did the header element in my previous implementation:
+            # from .oxml.header import CT_Hdr
+            # header_elm = CT_Hdr.new()
+            # not sure what is appropriate here for this first unit test
+            header_elm = 'pretend element'
+            self._header = Header(header_elm, self)
+
+        return self._header
```

Comment:
No need for a class variable here. This will just look like this:

```python
@lazyproperty
def header(self):
    """
    Return the |Header| object representing the default header for this
    section. A |Header| object is always returned, whether such a header is
    present or not. The header itself is added, updated, or removed using the
    returned object.
    """
    return Header(self._sectPr)
```

A few remarks:

* I order methods alphabetically within the class; that saves a full scan when looking for a particular one later.
* Every method gets its full docstring on creation. Same with classes. One never seems to get back to it later :)
* You'll need a Header class definition for this to pass. Which means we'll need a new file docx/header.py.